src/test/java/fi/metatavu/metaform/server/test/functional/tests/AdminThemeTestsIT.kt

```diff
@@ -0,0 +1,109 @@
+        QuarkusTestResource(MysqlResource::class),
+        QuarkusTestResource(KeycloakResource::class)
+)
+@TestProfile(GeneralTestProfile::class)
+class AdminThemeTestsIT : AbstractTest() {
+    @Test
+    @Throws(Exception::class)
+    fun createAdminThemeTest() {
+        TestBuilder().use { builder ->
+            val adminTheme = builder.metaformAdmin.adminThemes.create(
+                    data = "data",
+                    name = "Test admin theme",
+                    slug = "test-admin-theme"
+            )
+            
+            val foundAdminTheme = builder.metaformAdmin.adminThemes.findById(adminTheme.id!!)
+            Assertions.assertNotNull(foundAdminTheme)
+            Assertions.assertNotNull(foundAdminTheme.id)
+            Assertions.assertNotNull(foundAdminTheme.data)
+            Assertions.assertNotNull(foundAdminTheme.name)
+            Assertions.assertNotNull(foundAdminTheme.slug)
+        }
+    }
+
+    @Test
+    @Throws(Exception::class)
+    fun updateAdminThemeTest() {
+        TestBuilder().use { builder ->
+            val adminTheme = builder.metaformAdmin.adminThemes.create(
+                    data = "data",
+                    name = "Test admin theme",
+                    slug = "test-admin-theme"
+            )
+            val secondAdminTheme = builder.metaformAdmin.adminThemes.create(
+                    data = "data2",
+                    name = "Test admin theme2",
+                    slug = "test-admin-theme2"
+            )
+
+            val updatedAdminTheme = builder.metaformAdmin.adminThemes.update(adminTheme.id!!, secondAdminTheme)
+            val foundAdminTheme = builder.metaformAdmin.adminThemes.findById(adminTheme.id!!)
+            Assertions.assertNotNull(updatedAdminTheme)
+            Assertions.assertNotNull(updatedAdminTheme.id)
+            Assertions.assertNotNull(updatedAdminTheme.data)
+            Assertions.assertNotNull(updatedAdminTheme.name)
+            Assertions.assertNotNull(updatedAdminTheme.slug)
+            Assertions.assertEquals(updatedAdminTheme.data, foundAdminTheme.data)
+        }
+    }
+    
+    @Test
+    @Throws(Exception::class)
+    fun listAdminThemesTest() {
+        TestBuilder().use { builder ->
+            val adminTheme = builder.metaformAdmin.adminThemes.create(
+                    data = "data",
+                    name = "Test admin theme",
+                    slug = "test-admin-theme"
+            )
+            val secondAdminTheme = builder.metaformAdmin.adminThemes.create(
+                    data = "data2",
+                    name = "Test admin theme2",
+                    slug = "test-admin-theme2"
+            )
+
+            Assertions.assertNotNull(adminTheme)
+            Assertions.assertNotNull(secondAdminTheme)
+            val adminThemes = builder.metaformAdmin.adminThemes.list()
+            Assertions.assertNotNull(adminThemes)
+            Assertions.assertEquals(4, adminThemes.size)
```

Comment:
Why is here 4 given that you created only 2 themes?

Reply:
The test client should clean all the resources for each test. check the clean method in the test resource